pkg/metainfo/kvmetainfo/buckets.go

```diff
@@ -137,30 +169,67 @@ func (db *Project) ListBuckets(ctx context.Context, options storj.BucketListOpti
 		endBefore = "\x7f\x7f\x7f\x7f\x7f\x7f\x7f"
 	}
 
-	items, more, err := db.buckets.List(ctx, startAfter, endBefore, options.Limit)
+	objItems, more, err := db.buckets.List(ctx, "", startAfter, endBefore, false, options.Limit, meta.Modified)
 	if err != nil {
 		return storj.BucketList{}, err
 	}
 
 	list = storj.BucketList{
 		More:  more,
-		Items: make([]storj.Bucket, 0, len(items)),
+		Items: make([]storj.Bucket, 0, len(objItems)),
 	}
 
-	for _, item := range items {
-		list.Items = append(list.Items, bucketFromMeta(item.Bucket, item.Meta))
+	for _, itm := range objItems {
+		if itm.IsPrefix {
+			continue
+		}
+		m, err := convertMeta(ctx, itm.Path, itm.Meta)
+		if err != nil {
+			return storj.BucketList{}, err
+		}
+		list.Items = append(list.Items, m)
 	}
 
 	return list, nil
 }
 
-func bucketFromMeta(bucketName string, meta buckets.Meta) storj.Bucket {
-	return storj.Bucket{
-		Name:                 bucketName,
-		Created:              meta.Created,
-		PathCipher:           meta.PathEncryptionType,
-		SegmentsSize:         meta.SegmentsSize,
-		RedundancyScheme:     meta.RedundancyScheme,
-		EncryptionParameters: meta.EncryptionScheme.ToEncryptionParameters(),
-	}
+func convertMeta(ctx context.Context, bucketName string, m objects.Meta) (out storj.Bucket, err error) {
```

Comment:
bucketFromMeta was a slighty better name because there is also ObjectMeta

Reply:
done